docs/pages/api/automatically-register-agents.mdx

```diff
@@ -0,0 +1,1023 @@
+---
+title: Automatically Register Resources with Teleport
+description: Learn how to use the Teleport API to start agents automatically when you add resources to your infrastructure.
+---
+
+You can use Teleport's API to automatically register resources in your
+infrastructure with your Teleport cluster. 
+
+Teleport already supports the automatic discovery of [Kubernetes
+clusters](../kubernetes-access/discovery.mdx) in AWS, Azure, and Google Cloud,
+as well as [servers](../server-access/guides/ec2-discovery.mdx) on Amazon EC2.
+To support other resources and cloud providers, you can use the API to write
+your own workflow.
+
+In this guide, we will demonstrate some libraries you can use to automatically
+register resources with Teleport. We will use an example you can run locally on
+your workstation.
+
+Automatic registration consists of the following steps:
+
+- Look up resources in your infrastructure from a service discovery solution,
+  e.g., the Kubernetes API server, Consul, or your cloud provider's APIs.
+- Use the Teleport gRPC API to look up resources registered with Teleport.
+- For any resources in your infrastructure that are not registered with
+  Teleport, use the Teleport API to spin up a new Teleport service or register
+  the resource with an existing Teleport service.
+- For any resources that are registered with Teleport but not in your
+  infrastructure, use the Teleport API to deregister the resource and, if
+  necessary, remove the Teleport service proxying the resource.
+
+<Notice type="danger">
+
+The program we will build in this guide is intended as a learning tool. **Do not
```

Comment:
This might be over-jealousness on my part. I'm trying to follow the advice from [this talk](https://writing.rocks/to-be-or-not-to-be/), and the broader advice in @MarciaRieferJohnston's book. In this example, "will" is a "weak be verb" in that it's shorthand for "will be".  

In general, I'm always trying to reduce the amount of words needed to say something. I think about how I would say it if I were speaking aloud. It's probably that last point that most compels me to make suggestions like this.

It's your call buddy, do as you will.

Reply:
@alexfornuto Hi, Alex. This is a thoughtful suggestion, as always. Nice to be remembered. :) 

@ptgott Hi, Paul. Good question. Future tense ("will" + verb) isn't wrong. And you're right about the subject ("we") still performing the action with or without the "will." 

FWIW, here's one person's way of describing the reason for preferring present tense to future tense, which is generally considered a best practice (but not a have-to-do): 

> "In technical content, the future tense does not have the same immediacy as the present tense. Compare 'When you connect the power cable, the device will boot up automatically,' to 'When you connect the power cable, the device boots up automatically.' The ... present tense conveys immediacy and demonstrates confidence."

In sentences like yours, which sound fine either way (with or without the "will"), it's a judgment call. Tighter is generally better and stronger, so I'd leave it out. In cases where something will happen later and the relative time difference makes it weird to use present tense, though, then sure, use "will." As Alex says, it's up to you.

Reply:
Thanks for taking the time to contribute your thoughts!